forks/frames/src/main/java/com/tinkerpop/frames/modules/javahandler/JavaMethodHandler.java

```diff
@@ -2,40 +2,132 @@
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
 
 import com.tinkerpop.blueprints.Element;
 import com.tinkerpop.frames.FramedGraph;
 import com.tinkerpop.frames.util.ExceptionUtils;
 
 class JavaMethodHandler implements
-		com.tinkerpop.frames.modules.MethodHandler<JavaHandler> {
-
-	private JavaHandlerModule module;
-
-	JavaMethodHandler(JavaHandlerModule module) {
-		this.module = module;
-	}
-
-	@Override
-	public Class<JavaHandler> getAnnotationType() {
-		return JavaHandler.class;
-	}
-
-	@Override
-	public Object processElement(Object framedElement, Method method,
-			Object[] arguments, JavaHandler annotation,
-			FramedGraph<?> framedGraph, Element element) {
-		try {
-			Object handler = module.createHandler(framedElement, framedGraph, element, method.getDeclaringClass(), method);
-			return method.invoke(handler, arguments);
-		} catch (IllegalArgumentException e) {
-			throw new JavaHandlerException("Problem calling Java handler", e);
-		} catch (IllegalAccessException e) {
-			throw new JavaHandlerException("Problem calling Java handler", e);
-		} catch (InvocationTargetException e) {
-			ExceptionUtils.sneakyThrow(e.getTargetException());
-			return null;
-		}
-	}
+            com.tinkerpop.frames.modules.MethodHandler<JavaHandler>
+{
+
+    private JavaHandlerModule module;
+    private Map<ClassAndMethod, Class<?>> proxyToJavaHandlerCache = new HashMap<ClassAndMethod, Class<?>>();
+
+    JavaMethodHandler(JavaHandlerModule module)
+    {
+        this.module = module;
+    }
+
+    @Override
+    public Class<JavaHandler> getAnnotationType()
+    {
+        return JavaHandler.class;
+    }
+
+    @Override
+    public Object processElement(Object framedElement, Method method,
```

Comment:
Sure, it is because of the interface ToFileModelTranfosrmable that has a @JavaHandler method but does not have implementation